### PR TITLE
[libra framework] add TransactionChecks pre condition and specify …

### PR DESCRIPTION
**Pull request by emmazzz**
…more scripts

## Motivation

This PR
- adds `LibraAccount::TransactionChecks` schema specifying properties checked by the prologue
(currently there are just two: account exists and is not frozen),
- insert the schema mentioned above to all the specified scripts,
- adds specifications for `create_parent_vasp_account` and `create_designated_dealer` scripts, as well as corresponding functions in `LibraAccount`.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### CODE REVIEW

File: language/stdlib/modules/LibraAccount.move [hunk 1]
```diff
@@ -1422,6 +1491,17 @@ module LibraAccount {
         aborts_if !LibraTransactionPublishingOption::spec_is_script_allowed(sender, script_hash) with Errors::INVALID_STATE;
     }
 
+    spec define prologue_guarantees(sender: signer) : bool {
+        let addr = Signer::spec_address_of(sender);
+        LibraTimestamp::is_operating() && exists_at(addr) && !AccountFreezing::account_is_frozen(addr)
+    }
+
+    /// Used in transaction script to specify pre-conditions checked by the prologue.
+    spec schema TransactionCertified {
```
Comment (DavidLDill) [inline language/stdlib/modules/LibraAccount.move:1500]: Is "TransactionCertified" a good name?  I don't understand it.  PrologueGuarantees or something like that would be more immediately understandable.

**Review by DavidLDill — APPROVED** [review]

File: language/stdlib/modules/LibraAccount.move (hunk 1 above)
Comment (wrwg) [reply language/stdlib/modules/LibraAccount.move:1500]: What about `IsValidTransaction`? Emma and I discussed the "guarantees" yesterday, and it is also not to good. The `IsValid` seems to be the most neutral.

**Review by wrwg — APPROVED** [review]
Looks great modulo renaming of the schema. Let's try to land this quickly to avoid merge conflicts. I think @DavidLDill 's PR is in now.

**Pull request closed without merging**